BlogTemplate/Models/BlogDataStore.cs

```diff
@@ -217,7 +217,10 @@ public void AppendPostInfo(Post post, XElement rootNode)
 
         public void SavePost(Post post)
         {
-            SetId(post);
+            if (post.Id == 0)

```

Comment:
You could put this logic in the SetId method.  It makes that method solely responsible for making sure the ID is set correctly vs. having to determine here whether you need to call it.